sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -1037,10 +1037,11 @@ class SGDClassifier(BaseSGDClassifier):
 
     power_t : float, default=0.5
         The exponent for inverse scaling learning rate [default 0.5].
+        Values should be in the range `(-inf, inf)`.
```

Comment:
I do not anticipate negative `power_t` to be mathematically meaningful but apparently our code accepts it without crashing... So ok with documenting it.

Reply:
I feel like this a case where documenting `-inf` will lead to more people trying out. If this is not mathematically meaningful, then we could be promoting a bad practice?

Reply:
I tested the code in the Binder notebook. 
We can open up an issue to address this.  Or, is this something that can be set up with tests in the next step, which will be adding `check_scalar` functions?

Reply:
Opened issue #22178